pkg/sql/region_util.go

```diff
@@ -375,11 +383,24 @@ func zoneConfigForMultiRegionTable(
 		// nothing here because `NewZoneConfig()` already marks those fields as
 		// 'inherited'.
 	case *descpb.TableDescriptor_LocalityConfig_RegionalByTable_:
-		// Use the same configuration as the database and return nil here.
+		restrictedPlacement := regionConfig.Placement() == descpb.DataPlacement_RESTRICTED
```

Comment:
my comment from the earlier PR applies here; should we make the database descriptor have the restricted placement by default, and instead have global tables with DEFAULT placement strictness specify voters? this is prone to errors.

https://reviewable.io/reviews/cockroachdb/cockroach/67946#-MfGOVrlBMLqHltKbHRS

Reply:
Agreed. Just pushed this up.